### ISSUE TITLE
Upload errors

### ISSUE THREAD

**Issue by Wrascon** [gustavollps/esp32TTGO_i2s_scope#3]
Hello, i use this parameters to try upload your sketch but it fails.
Arduino: 1.8.13 (Windows 10), Плата:"TTGO LoRa32-OLED V1, 80MHz, 921600, None"
At first time i see error which didsnt find Button2.h, I put this file in folder and included with "" in I2S_adc_esp32.ino.
After that file i2s.ino give me int error at line 16:
"narrowing conversion of 'rate' from 'uint32_t {aka unsigned int}' to 'int' inside { } [-Werror=narrowing]".
What i need to do with libraries to upload your Oscilloscope sketch on my TTGO too?

**Comment by gustavollps** [comment]
Hi, Wrascon. I'm gonna try to reproduce your error here, but the Button2 library you can install from the Arduino IDE library manager. Try it for now and see if it fix your error. About the uint32_t on the i2c.ino I don't have a clue yet. Any updates I will let you know here.

**Comment by gustavollps** [comment]
Tested here and everything looks ok, with version 1.8.13.
To successful compile the code, you should add the libraries (by the IDE's library manager):
- TFT_eSPI (by Bodmer) - compiled for me with version 2.3.59
- Button2 - version 1.5.0

With the libraries added, I did not get any error related to the uint32_t. Have you already added the ESP32 board to your IDE's board manager?
Select the ESP32 Wrover module as the board and upload with the default parameters (Upload speed, flash frequency, etc), it should work fine.

**Comment by gustavollps** [comment]
Ahh, you should also change the TFT_eSPI/User_Setup_Select.h file on the TFT library to use the correct display, as described here:
https://oneguyoneblog.com/2020/09/28/lilygo-ttgo-t-display-esp32-with-tft-lcd/

**Comment by Wrascon** [comment]
Hello. I already change my User_Setup_Select.h file as they say in this guide, also add this board link to choose ESP32 Arduino list. After download library Button2 I can remove my Button2.h from directory this sketch for avoid first include error. But I stuck on "narrowing conversion" again in i2s.ino file. Maybe I use other library, but after clean some of them nothing happens to start this correctly.
https://pastebin.com/9VXzn1j5
![image](https://user-images.githubusercontent.com/17739216/105649270-9b780d00-5ec0-11eb-8901-3921c6c96d5f.png)

**Comment by gustavollps** [comment]
Sorry for the time it took me to answer.
I figured out that you are using the "compiler warnings" option, and I was not. This gave you the error that I was not seeing here.
I am going to update the code to remove the warnings and make it more reliable.
The error is related to converting an uint32_t to an int that is also 32 bit on esp32 (I did not know, as Arduino standard int is 16bit). 

I reformated the code to use a standard int. You should have no problems now. I hope.

**Comment by Wrascon** [comment]
Hello. I tried to upload your new sketch, it didn't show any errors, but nothing is displayed on the screen. This didn't stop it from showing the final download statistics. Maybe I have some incorrect libraries or in addition to the power from the USB to the board you need to connect something else? This confuses me and it is not yet clear what is causing the error on what would seem to be the same board on my part. As I see here is some warnings with "int i2s_read_bytes" same as came before previous error on screenshot. Although I may be wrong, the error occurs at this stage of compiling the sketch, and not the libraries themselves.
![image](https://user-images.githubusercontent.com/17739216/106023085-888b5580-60d7-11eb-8016-69a7a5e6757b.png)

<details>
  <summary>error</summary>

  ```
D:\Program Files\Arduino\PROJECTS\esp32TTGO_i2s_scope-master\I2S_adc_esp32\i2s.ino: In function 'void ADC_Sampling(uint16_t*)':
D:\Program Files\Arduino\PROJECTS\esp32TTGO_i2s_scope-master\I2S_adc_esp32\i2s.ino:29:5: warning: 'int i2s_read_bytes(i2s_port_t, void*, size_t, TickType_t)' is deprecated [-Wdeprecated-declarations]
     i2s_read_bytes(I2S_NUM_0, (char*)&i2s_buff[i * NUM_SAMPLES],  NUM_SAMPLES * sizeof(uint16_t), portMAX_DELAY);    
     ^
In file included from D:\Program Files\Arduino\PROJECTS\esp32TTGO_i2s_scope-master\I2S_adc_esp32\I2S_adc_esp32.ino:2:0:
C:\Users\zashi\AppData\Local\Arduino15\packages\esp32\hardware\esp32\1.0.4/tools/sdk/include/driver/driver/i2s.h:341:5: note: declared here
 int i2s_read_bytes(i2s_port_t i2s_num, void *dest, size_t size, TickType_t ticks_to_wait) __attribute__ ((deprecated));
     ^
D:\Program Files\Arduino\PROJECTS\esp32TTGO_i2s_scope-master\I2S_adc_esp32\i2s.ino:29:5: warning: 'int i2s_read_bytes(i2s_port_t, void*, size_t, TickType_t)' is deprecated [-Wdeprecated-declarations]
     i2s_read_bytes(I2S_NUM_0, (char*)&i2s_buff[i * NUM_SAMPLES],  NUM_SAMPLES * sizeof(uint16_t), portMAX_DELAY);    
     ^
In file included from D:\Program Files\Arduino\PROJECTS\esp32TTGO_i2s_scope-master\I2S_adc_esp32\I2S_adc_esp32.ino:2:0:
C:\Users\zashi\AppData\Local\Arduino15\packages\esp32\hardware\esp32\1.0.4/tools/sdk/include/driver/driver/i2s.h:341:5: note: declared here
 int i2s_read_bytes(i2s_port_t i2s_num, void *dest, size_t size, TickType_t ticks_to_wait) __attribute__ ((deprecated));
     ^
D:\Program Files\Arduino\PROJECTS\esp32TTGO_i2s_scope-master\I2S_adc_esp32\i2s.ino:29:112: warning: 'int i2s_read_bytes(i2s_port_t, void*, size_t, TickType_t)' is deprecated [-Wdeprecated-declarations]
     i2s_read_bytes(I2S_NUM_0, (char*)&i2s_buff[i * NUM_SAMPLES],  NUM_SAMPLES * sizeof(uint16_t), portMAX_DELAY);    
                                                                                                                ^
In file included from D:\Program Files\Arduino\PROJECTS\esp32TTGO_i2s_scope-master\I2S_adc_esp32\I2S_adc_esp32.ino:2:0:
C:\Users\zashi\AppData\Local\Arduino15\packages\esp32\hardware\esp32\1.0.4/tools/sdk/include/driver/driver/i2s.h:341:5: note: declared here
 int i2s_read_bytes(i2s_port_t i2s_num, void *dest, size_t size, TickType_t ticks_to_wait) __attribute__ ((deprecated));
     ^
  ```
</details>

The sketch uses 364,501 bytes (27%) of the device's memory. A total of 1310,720 bytes are available.
Global variables use 117588 bytes (35%) of dynamic memory, leaving 210092 bytes for local variables. Maximum: 327680 bytes.
![image](https://user-images.githubusercontent.com/17739216/106021043-714b6880-60d5-11eb-9a00-532d94dba2ad.png)

<details>
  <summary>result</summary>

  ```
esptool.py v2.6
Serial port COM18
Connecting.......
Chip is ESP32D0WDQ6 (revision 1)
Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
MAC: 
Uploading stub...
Running stub...
Stub running...
Changing baud rate to 921600
Changed.
Configuring flash size...
Auto-detected Flash size: 4MB
Compressed 8192 bytes to 47...
Wrote 8192 bytes (47 compressed) at 0x0000e000 in 0.0 seconds (effective 16384.2 kbit/s)...
Hash of data verified.
Compressed 15856 bytes to 10276...
Wrote 15856 bytes (10276 compressed) at 0x00001000 in 0.1 seconds (effective 925.9 kbit/s)...
Hash of data verified.
Compressed 314112 bytes to 163100...
Wrote 314112 bytes (163100 compressed) at 0x00010000 in 2.7 seconds (effective 924.9 kbit/s)...
Hash of data verified.
Compressed 3072 bytes to 128...
Wrote 3072 bytes (128 compressed) at 0x00008000 in 0.0 seconds (effective 4915.3 kbit/s)...
Hash of data verified.

Leaving...
Hard resetting via RTS pin...
  ```
</details>

**Comment by gustavollps** [comment]
Yes, this "warning" is related to a deprecated method that will not be maintained in the future and is not necessary (probably because there is a new one to replace it), but still works, therefore it remains on the core of the esp32.

You could try the official sample code for the board to test the display:
https://github.com/Xinyuan-LilyGO/TTGO-T-Display
If this code works, replace the TFT library used on the scope code with the library provided on the TTGO repository.

You can also open the serial monitor with the board connected: you should see the CORE0 CORE1 prints, which means the code is running, just the display is not working.

The next time I get some free time I will fix this deprecated warning more calmly.

**Comment by Wrascon** [comment]
Thanks. I can finally use it. After some failed tests, I was able to figure out that my board went into a deep loop and I had to load something simple with the reset button held down. Also, due to my inattention, the settings of the standard display were changed in one library, but Arduino thought it best to choose the second one that was not configured. I realized this from the errors of busy Wifi, BT, and Serial that are similar to this problem.
